### PR TITLE
Add customer controller unit tests

### PR DESCRIPTION
**Pull request by tsukimi2**
Add customer controller unit test

### CODE REVIEW

File: server/controllers/customer.spec.js [hunk 1]
```diff
@@ -0,0 +1,458 @@
+const sinon = require('sinon')
+const sandbox = sinon.createSandbox()
+import {omit} from 'lodash'
+const customerCtrl = require('./customer').default
```
Comment (jspaine) [inline server/controllers/customer.spec.js]: Why not just import this and sinon instead of requiring?

File: server/controllers/customer.spec.js [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+          notifications: [ user.notifications[0] ],
+          id: user._id,
+          email: user.email,
+          provider: user.provider,
+        }
+      }
+
+      res = {
+        json: sandbox.spy(),
+        status: sandbox.stub().returns({ end: sandbox.spy() })
```
Comment (jspaine) [inline server/controllers/customer.spec.js]: Nothing calls res.status().end() so this isn't needed. Sinon spies/stubs have a `returnsThis` that can be useful:
```
res = {
  status: sandbox.stub().returnsThis,
  end: sandbox.spy()
}
```

File: server/controllers/customer.spec.js [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+describe('Customer controller', function() {
+  describe('create', function() {
+    let req = {}
+    let res = {}
+    let user = null
+    let newCustomer = null
+    let userSession = null
+
+    before(async function() {
```
Comment (jspaine) [inline server/controllers/customer.spec.js]: I'd try to get rid of these, setting up the db shouldn't be needed for a unit test if everything is stubbed.

File: server/controllers/customer.spec.js [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+    after(async function() {
+      await Questionnaire.find().remove()
+      await resetDb()
+      user = null
+    })
+
+    beforeEach(function() {
+      req = {
+        params: {},
+        body: {
```
Comment (jspaine) [inline server/controllers/customer.spec.js:20]: Most of `req.body`, `req.user` and `expectedResult` doesn't seem to be needed, I'd cut it down to only what the test is looking for.

File: server/controllers/customer.spec.js [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+        household: [],
+        fields: [],
+        _id: 45633,
+        firstName: 'user',
+        lastName: 'test',
+        email: 'user@test.com'
+      }]
+
+      mockFindAll = {
+        sort: function() {
```
Comment (jspaine) [inline server/controllers/customer.spec.js]: Can simplify with `returnsThis` - `sort: sinon.stub().returnsThis`

**Review by jspaine — COMMENTED** [review]
Thanks, it looks good. I'd try to cut down on the mock data though and get rid of the db setup (if everything's stubbed nothing should be calling the db)

File: server/controllers/customer.spec.js [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+      }
+    })
+
+    afterEach(function() {
+      req = {}
+      res = {}
+      sandbox.restore()
+    })
+
+    it('should update customer and volunteer and return deleted customer object if the customer is associated with any packages', async function() {
```
Comment (jspaine) [inline server/controllers/customer.spec.js]: I'd nest these so it's not "it should do all the things", something like:
```
describe("when the customer has packages")
  it("should update the customer")
  it("should update the volunteer")
  it("should return the deleted customer")
describe("when the customer doesnt have packages")
  it("should respond 409")
```

**Comment by tsukimi2** [comment]
Updated pull request with suggestions you made above.   However, after I pushed the amended commit up the repos, the Travis CI build failed, with the test for User Api Signup "1) does not allow signup if email address already has an account" failing.  When I run the test on my local repos, all the tests passed without problem.  Would you be kind enough to take a look at this problem? Thanks.

**Pull request closed without merging**